### DISCUSSION TITLE
Unable to Select Organization During User Login

### DISCUSSION THREAD

**Discussion by arbha1erao** [keycloak/keycloak#33911] [Q&A]
I’ve created two organizations, and I have a user that is part of both. However, when I try to log in with this user, I don’t see an option to select which organization to log into.

Has anyone encountered this issue? Any insights or solutions would be greatly appreciated!

Thanks in advance!

**Comment by kelvinsleonardo** [comment]
@arbha1erao Any solution?

**Reply by arbha1erao** [reply]
Did not find any solution yet.

**Reply by kelvinsleonardo** [reply]
I found the solution https://github.com/keycloak/keycloak/discussions/35581

**Comment by mildlypranav** [comment]
Hey — we ran into the same gap and built an open-source Keycloak Authenticator SPI that handles this:

**https://github.com/revspot/keycloak-select-org-authenticator**

It plugs into the browser authentication flow as a post-login step and queries the user's **native Keycloak 26+ Organization memberships**:
- 0 orgs → denies access
- 1 org → auto-selects silently
- 2+ orgs → shows a picker page

Stores the selection in session notes (`kc.org`) so the built-in Organization Membership token mapper picks it up automatically. Hope it helps!
